In [26]:
# Loading the required packages for loading the dataset
import pandas as pd # Loading the dataset 
import numpy as np # applying any mathematical operations on it


In [27]:
# Loading the dataset 
df = pd.read_csv(r'D:\Final submission\EDA\COVID19-analysis-EDA\Dataset\country_wise_latest.csv')
df

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  \
0           Afghanistan      36263    1269      25198    9796        106   
1               Albania       4880     144       2745    1991        117   
2               Algeria      27973    1163      18837    7973        616   
3               Andorra        907      52        803      52         10   
4                Angola        950      41        242     667         18   
..                  ...        ...     ...        ...     ...        ...   
182  West Bank and Gaza      10621      78       3752    6791        152   
183      Western Sahara         10       1          8       1          0   
184               Yemen       1691     483        833     375         10   
185              Zambia       4552     140       2815    1597         71   
186            Zimbabwe       2704      36        542    2126        192   

     New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0            10             18                3.50                  69.49   
1             6             63                2.95                  56.25   
2             8            749                4.16                  67.34   
3             0              0                5.73                  88.53   
4             1              0                4.32                  25.47   
..          ...            ...                 ...                    ...   
182           2              0                0.73                  35.33   
183           0              0               10.00                  80.00   
184           4             36               28.56                  49.26   
185           1            465                3.08                  61.84   
186           2             24                1.33                  20.04   

     Deaths / 100 Recovered  Confirmed last week  1 week change  \
0                      5.04                35526            737   
1                      5.25                 4171            709   
2                      6.17                23691           4282   
3                      6.48                  884             23   
4                     16.94                  749            201   
..                      ...                  ...            ...   
182                    2.08                 8916           1705   
183                   12.50                   10              0   
184                   57.98                 1619             72   
185                    4.97                 3326           1226   
186                    6.64                 1713            991   

     1 week % increase             WHO Region  
0                 2.07  Eastern Mediterranean  
1                17.00                 Europe  
2                18.07                 Africa  
3                 2.60                 Europe  
4                26.84                 Africa  
..                 ...                    ...  
182              19.12  Eastern Mediterranean  
183               0.00                 Africa  
184               4.45  Eastern Mediterranean  
185              36.86                 Africa  
186              57.85                 Africa  

[187 rows x 15 columns]

## Description of data
* A new coronavirus designated 2019-nCoV was first identified in Wuhan, the capital of China's Hubei province
* People developed pneumonia without a clear cause and for which existing vaccines or treatments were not effective.
* The virus has shown evidence of human-to-human transmission
* Transmission rate (rate of infection) appeared to escalate in mid-January 2020
* As of 30 January 2020, approximately 8,243 cases have been confirmed

Columns :

1. Country/Region : This cloumn indicate the country or regions name.
2. Confirmed : This column represent the confirmed cases.
3. Deaths : This column represent the death cases.
4. Recovered : This column represent the information regarding the recorved person.
5. Active : Number of active cases.
6. New cases : Number of new cases.
7. New deaths : New deaths cases.
8. New recovered : Newly recovered.
9. Deaths / 100 Cases : Count of death for every 100 people.
10. Recovered / 100 Cases : Recorved by 100 people
11. Deaths / 100 Recovered :
12. Confirmed last week : Cases that are confirmed in last week
13. 1 week change : changes in first week
14. 1 week % increase : percetage of cases increased in 1 week
15. WHO Region : Region name

## Steps needed to be followed:
1. check the information and know its dtype.
2. Checking null values if the values are persent then replace it.
3. Describe method to get stat understanding.
4. Identify the unique elements for all columns.
5. Check for improper string and replace it with proper string.
6. Cross check all the above steps



In [28]:
# Step 1 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country/Region          187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    float64
 9   Recovered / 100 Cases   187 non-null    float64
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    float64
 14  WHO Region              187 non-null    ob

From the above we can see that all the dtype of my data is in good format.

In [29]:
# Step 2
df.isnull().sum()

Country/Region            0
Confirmed                 0
Deaths                    0
Recovered                 0
Active                    0
New cases                 0
New deaths                0
New recovered             0
Deaths / 100 Cases        0
Recovered / 100 Cases     0
Deaths / 100 Recovered    0
Confirmed last week       0
1 week change             0
1 week % increase         0
WHO Region                0
dtype: int64

From the above we can see that there are no null value persent in my dataset.

In [30]:
# Step 3

df.describe()

Confirmed         Deaths     Recovered        Active     New cases  \
count  1.870000e+02     187.000000  1.870000e+02  1.870000e+02    187.000000   
mean   8.813094e+04    3497.518717  5.063148e+04  3.400194e+04   1222.957219   
std    3.833187e+05   14100.002482  1.901882e+05  2.133262e+05   5710.374790   
min    1.000000e+01       0.000000  0.000000e+00  0.000000e+00      0.000000   
25%    1.114000e+03      18.500000  6.265000e+02  1.415000e+02      4.000000   
50%    5.059000e+03     108.000000  2.815000e+03  1.600000e+03     49.000000   
75%    4.046050e+04     734.000000  2.260600e+04  9.149000e+03    419.500000   
max    4.290259e+06  148011.000000  1.846641e+06  2.816444e+06  56336.000000   

        New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
count   187.000000     187.000000          187.000000             187.000000   
mean     28.957219     933.812834            3.019519              64.820535   
std     120.037173    4197.719635            3.454302              26.287694   
min       0.000000       0.000000            0.000000               0.000000   
25%       0.000000       0.000000            0.945000              48.770000   
50%       1.000000      22.000000            2.150000              71.320000   
75%       6.000000     221.000000            3.875000              86.885000   
max    1076.000000   33728.000000           28.560000             100.000000   

       Deaths / 100 Recovered  Confirmed last week  1 week change  \
count                  187.00         1.870000e+02     187.000000   
mean                      inf         7.868248e+04    9448.459893   
std                       NaN         3.382737e+05   47491.127684   
min                      0.00         1.000000e+01     -47.000000   
25%                      1.45         1.051500e+03      49.000000   
50%                      3.62         5.020000e+03     432.000000   
75%                      6.44         3.708050e+04    3172.000000   
max                       inf         3.834677e+06  455582.000000   

       1 week % increase  
count         187.000000  
mean           13.606203  
std            24.509838  
min            -3.840000  
25%             2.775000  
50%             6.890000  
75%            16.855000  
max           226.320000

From the above we can see that there are few rows represent the following:--

1. count of the data
2. mean of every data
3. std - deivation from mean

### Five number summary
4. min 
5. 25%  
6. 50%
7. 75%
8. max

In [31]:
# let's check the column name list
df.columns

Index(['Country/Region', 'Confirmed', 'Deaths', 'Recovered', 'Active',
       'New cases', 'New deaths', 'New recovered', 'Deaths / 100 Cases',
       'Recovered / 100 Cases', 'Deaths / 100 Recovered',
       'Confirmed last week', '1 week change', '1 week % increase',
       'WHO Region'],
      dtype='object')

In [32]:
# step 4 checking the unique elemnet and its length
print(df['Country/Region'].unique())
print('')
print(len(df['Country/Region'].unique()))

['Afghanistan' 'Albania' 'Algeria' 'Andorra' 'Angola'
 'Antigua and Barbuda' 'Argentina' 'Armenia' 'Australia' 'Austria'
 'Azerbaijan' 'Bahamas' 'Bahrain' 'Bangladesh' 'Barbados' 'Belarus'
 'Belgium' 'Belize' 'Benin' 'Bhutan' 'Bolivia' 'Bosnia and Herzegovina'
 'Botswana' 'Brazil' 'Brunei' 'Bulgaria' 'Burkina Faso' 'Burma' 'Burundi'
 'Cabo Verde' 'Cambodia' 'Cameroon' 'Canada' 'Central African Republic'
 'Chad' 'Chile' 'China' 'Colombia' 'Comoros' 'Congo (Brazzaville)'
 'Congo (Kinshasa)' 'Costa Rica' "Cote d'Ivoire" 'Croatia' 'Cuba' 'Cyprus'
 'Czechia' 'Denmark' 'Djibouti' 'Dominica' 'Dominican Republic' 'Ecuador'
 'Egypt' 'El Salvador' 'Equatorial Guinea' 'Eritrea' 'Estonia' 'Eswatini'
 'Ethiopia' 'Fiji' 'Finland' 'France' 'Gabon' 'Gambia' 'Georgia' 'Germany'
 'Ghana' 'Greece' 'Greenland' 'Grenada' 'Guatemala' 'Guinea'
 'Guinea-Bissau' 'Guyana' 'Haiti' 'Holy See' 'Honduras' 'Hungary'
 'Iceland' 'India' 'Indonesia' 'Iran' 'Iraq' 'Ireland' 'Israel' 'Italy'
 'Jamaica' 'Japan' 'Jordan' '

From the above my observation is the there all 187 different country/regions but only one of the name is not in 
proper order i.e. 'Taiwan*' and my task is to make it in order.

In [33]:
df['Country/Region'] = df['Country/Region'].replace(['Taiwan*'],['Taiwan'])

In [34]:
# final check for Country/Region
df['Country/Region'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Andorra', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Australia',
       'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain', 'Bangladesh',
       'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin', 'Bhutan',
       'Bolivia', 'Bosnia and Herzegovina', 'Botswana', 'Brazil',
       'Brunei', 'Bulgaria', 'Burkina Faso', 'Burma', 'Burundi',
       'Cabo Verde', 'Cambodia', 'Cameroon', 'Canada',
       'Central African Republic', 'Chad', 'Chile', 'China', 'Colombia',
       'Comoros', 'Congo (Brazzaville)', 'Congo (Kinshasa)', 'Costa Rica',
       "Cote d'Ivoire", 'Croatia', 'Cuba', 'Cyprus', 'Czechia', 'Denmark',
       'Djibouti', 'Dominica', 'Dominican Republic', 'Ecuador', 'Egypt',
       'El Salvador', 'Equatorial Guinea', 'Eritrea', 'Estonia',
       'Eswatini', 'Ethiopia', 'Fiji', 'Finland', 'France', 'Gabon',
       'Gambia', 'Georgia', 'Germany', 'Ghana', 'Greece', 'Greenland',
       'Grenada', 'Guatemala', 'Guinea', 'G

In [35]:
df.head()

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
0    Afghanistan      36263    1269      25198    9796        106          10   
1        Albania       4880     144       2745    1991        117           6   
2        Algeria      27973    1163      18837    7973        616           8   
3        Andorra        907      52        803      52         10           0   
4         Angola        950      41        242     667         18           1   

   New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0             18                3.50                  69.49   
1             63                2.95                  56.25   
2            749                4.16                  67.34   
3              0                5.73                  88.53   
4              0                4.32                  25.47   

   Deaths / 100 Recovered  Confirmed last week  1 week change  \
0                    5.04                35526            737   
1                    5.25                 4171            709   
2                    6.17                23691           4282   
3                    6.48                  884             23   
4                   16.94                  749            201   

   1 week % increase             WHO Region  
0               2.07  Eastern Mediterranean  
1              17.00                 Europe  
2              18.07                 Africa  
3               2.60                 Europe  
4              26.84                 Africa

In [36]:
df['Confirmed'].nunique() 

184

From the above we can see that there are 184 unique confirmed cases from 187 different Country/Region.

In [37]:
df['Deaths'].nunique()

150

From the above we can see that there are 150 unique deaths cases from 187 different Country/Region.

In [38]:
df['Recovered'].nunique()

178

From the above we can see that there are 178 unique recorved cases from 187 different Country/Region.

In [39]:
df['Active'].nunique()

173

From the above we can see that there are 173 unique active cases from 187 different Country/Region.

In [40]:
df['New cases'].nunique()

122

From the above we can see that there are 122 unique new cases from 187 different Country/Region.

In [41]:
df.columns

Index(['Country/Region', 'Confirmed', 'Deaths', 'Recovered', 'Active',
       'New cases', 'New deaths', 'New recovered', 'Deaths / 100 Cases',
       'Recovered / 100 Cases', 'Deaths / 100 Recovered',
       'Confirmed last week', '1 week change', '1 week % increase',
       'WHO Region'],
      dtype='object')

In [42]:
df['New deaths'].nunique()

38

From the above we can see that there are 38 unique new death cases from 187 different Country/Region.

In [43]:
df['New recovered'].nunique()

103

From the above we can see that there are 103 unique newly recorved cases from 187 different Country/Region.

In [44]:
df['Deaths / 100 Cases'].nunique()

145

From the above we can see that there are 145 death per 100 cases from 187 different Country/Region.

In [45]:
df['Recovered / 100 Cases'].nunique()

177

From the above we can see that there are 177 recorved per 100 cases from 187 different Country/Region.

In [46]:
df['Deaths / 100 Recovered'].nunique()

155

From the above we can see that there are 155 death per 100 recovered from 187 different Country/Region.

In [47]:
df['Confirmed last week'].nunique()

183

From the above we can see that there are 183 confirmed last week cases from 187 different Country/Region.

In [48]:
df['1 week change'].nunique()

162

From the above we can see that there are 162 confirmed 1 week change  from 187 different Country/Region.

In [49]:
df['WHO Region'].value_counts()

Europe                   56
Africa                   48
Americas                 35
Eastern Mediterranean    22
Western Pacific          16
South-East Asia          10
Name: WHO Region, dtype: int64

From the above we can see that there are total 6 different regions and there count.

In [50]:
df.head()

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
0    Afghanistan      36263    1269      25198    9796        106          10   
1        Albania       4880     144       2745    1991        117           6   
2        Algeria      27973    1163      18837    7973        616           8   
3        Andorra        907      52        803      52         10           0   
4         Angola        950      41        242     667         18           1   

   New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0             18                3.50                  69.49   
1             63                2.95                  56.25   
2            749                4.16                  67.34   
3              0                5.73                  88.53   
4              0                4.32                  25.47   

   Deaths / 100 Recovered  Confirmed last week  1 week change  \
0                    5.04                35526            737   
1                    5.25                 4171            709   
2                    6.17                23691           4282   
3                    6.48                  884             23   
4                   16.94                  749            201   

   1 week % increase             WHO Region  
0               2.07  Eastern Mediterranean  
1              17.00                 Europe  
2              18.07                 Africa  
3               2.60                 Europe  
4              26.84                 Africa

In [51]:
df['Deaths / 100 Cases'] = df['Deaths / 100 Cases']*100
df['Deaths / 100 Cases']

0       350.0
1       295.0
2       416.0
3       573.0
4       432.0
        ...  
182      73.0
183    1000.0
184    2856.0
185     308.0
186     133.0
Name: Deaths / 100 Cases, Length: 187, dtype: float64

In [52]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country/Region          187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    float64
 9   Recovered / 100 Cases   187 non-null    float64
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    float64
 14  WHO Region              187 non-null    ob

In [53]:
df['Deaths / 100 Cases'] = df['Deaths / 100 Cases'].astype('int64')

In [54]:
df['Recovered / 100 Cases'] = df['Recovered / 100 Cases'].astype('int64')

In [55]:
df.columns

Index(['Country/Region', 'Confirmed', 'Deaths', 'Recovered', 'Active',
       'New cases', 'New deaths', 'New recovered', 'Deaths / 100 Cases',
       'Recovered / 100 Cases', 'Deaths / 100 Recovered',
       'Confirmed last week', '1 week change', '1 week % increase',
       'WHO Region'],
      dtype='object')

In [56]:
df['1 week % increase'] = df['1 week % increase'].astype('int64')

In [57]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country/Region          187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    int64  
 9   Recovered / 100 Cases   187 non-null    int64  
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    int64  
 14  WHO Region              187 non-null    ob

In [58]:
df.describe()

Confirmed         Deaths     Recovered        Active     New cases  \
count  1.870000e+02     187.000000  1.870000e+02  1.870000e+02    187.000000   
mean   8.813094e+04    3497.518717  5.063148e+04  3.400194e+04   1222.957219   
std    3.833187e+05   14100.002482  1.901882e+05  2.133262e+05   5710.374790   
min    1.000000e+01       0.000000  0.000000e+00  0.000000e+00      0.000000   
25%    1.114000e+03      18.500000  6.265000e+02  1.415000e+02      4.000000   
50%    5.059000e+03     108.000000  2.815000e+03  1.600000e+03     49.000000   
75%    4.046050e+04     734.000000  2.260600e+04  9.149000e+03    419.500000   
max    4.290259e+06  148011.000000  1.846641e+06  2.816444e+06  56336.000000   

        New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
count   187.000000     187.000000          187.000000             187.000000   
mean     28.957219     933.812834          301.903743              64.336898   
std     120.037173    4197.719635          345.441702              26.225631   
min       0.000000       0.000000            0.000000               0.000000   
25%       0.000000       0.000000           94.500000              48.500000   
50%       1.000000      22.000000          215.000000              71.000000   
75%       6.000000     221.000000          387.500000              86.000000   
max    1076.000000   33728.000000         2856.000000             100.000000   

       Deaths / 100 Recovered  Confirmed last week  1 week change  \
count                  187.00         1.870000e+02     187.000000   
mean                      inf         7.868248e+04    9448.459893   
std                       NaN         3.382737e+05   47491.127684   
min                      0.00         1.000000e+01     -47.000000   
25%                      1.45         1.051500e+03      49.000000   
50%                      3.62         5.020000e+03     432.000000   
75%                      6.44         3.708050e+04    3172.000000   
max                       inf         3.834677e+06  455582.000000   

       1 week % increase  
count         187.000000  
mean           13.181818  
std            24.512211  
min            -3.000000  
25%             2.000000  
50%             6.000000  
75%            16.500000  
max           226.000000

In [59]:
df.drop(columns=['Deaths / 100 Recovered'], axis=1,inplace=True)

In [60]:
df.head()

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
0    Afghanistan      36263    1269      25198    9796        106          10   
1        Albania       4880     144       2745    1991        117           6   
2        Algeria      27973    1163      18837    7973        616           8   
3        Andorra        907      52        803      52         10           0   
4         Angola        950      41        242     667         18           1   

   New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0             18                 350                     69   
1             63                 295                     56   
2            749                 416                     67   
3              0                 573                     88   
4              0                 432                     25   

   Confirmed last week  1 week change  1 week % increase  \
0                35526            737                  2   
1                 4171            709                 17   
2                23691           4282                 18   
3                  884             23                  2   
4                  749            201                 26   

              WHO Region  
0  Eastern Mediterranean  
1                 Europe  
2                 Africa  
3                 Europe  
4                 Africa

In [77]:
df.shape

(187, 14)

## Question that are solved in this notebook:


1. Based on each country plot what are the confirmed cases
2. Based on each country plot what are the deaths cases
3. Based on each country plot what are the Recovered cases
4. Find out the count of activa cases based on specific country
5. Top 10 confirmed cases in each country/region
6. New recovered count based on country
7. Top countries death ration
8. Finding the count of Deaths /100 Cases based on the country
11. Recorved cases based on Country/Region
10. Recoved cases based on WHO region

In [61]:
# Dependencies

import matplotlib.pyplot as plt
import seaborn as sns
import plotly as py                
import cufflinks as cf
%matplotlib inline
from plotly.offline import iplot   
py.offline.init_notebook_mode(connected=True)
cf.go_offline()
import plotly.express as px


In [78]:
# Based on each country plot what are the confirmed cases
fig = px.scatter(df[:100],x='Country/Region', y='Confirmed',color='WHO Region')
fig.show()

From the above we can see that WHO region Americas have more number of cases compare to all and in this plot i have 
considered Country/Region with confirmd cases.

In [63]:
#  Based on each country plot what are the deaths cases
fig = px.bar(df,x='Country/Region', y='Deaths',color='WHO Region',height=400)
fig.show()

From the above we can see that THe count of death durning the covid in every region.

In [64]:
# Based on each country plot what are the Recovered cases

fig = px.line(df[:100],x='Country/Region', y='Recovered',color='WHO Region')
fig.show()

From the above we can see how the recovery from covid has occour in every region.

In [65]:
# Find out the count of activa cases based on specific country
fig = px.scatter(df[:30],x='Country/Region', y='Active',color='WHO Region')
fig.show()

From the above we can see the how the count of active cases are increaing.

In [66]:
# Creating a new data frame and exploring it.

new_df = df[["Country/Region","Confirmed","Deaths","Recovered"]]
new_df
covid_df = new_df.sort_values(by="Confirmed",ascending=False)
covid_df = covid_df.reset_index(drop=True)
covid_df

Country/Region  Confirmed  Deaths  Recovered
0                       US    4290259  148011    1325804
1                   Brazil    2442375   87618    1846641
2                    India    1480073   33408     951166
3                   Russia     816680   13334     602249
4             South Africa     452529    7067     274925
..                     ...        ...     ...        ...
182               Dominica         18       0         18
183  Saint Kitts and Nevis         17       0         15
184              Greenland         14       0         13
185               Holy See         12       0         12
186         Western Sahara         10       1          8

[187 rows x 4 columns]

In [67]:
# Top 10 confirmed cases in country

top_10 = df[['Country/Region','Confirmed']].value_counts()

In [68]:
covid_df[:10]

Country/Region  Confirmed  Deaths  Recovered
0              US    4290259  148011    1325804
1          Brazil    2442375   87618    1846641
2           India    1480073   33408     951166
3          Russia     816680   13334     602249
4    South Africa     452529    7067     274925
5          Mexico     395489   44022     303810
6            Peru     389717   18418     272547
7           Chile     347923    9187     319954
8  United Kingdom     301708   45844       1437
9            Iran     293606   15912     255144

In [69]:
fig = px.bar(covid_df[:10], x = 'Country/Region', y= 'Confirmed',title="Top 10 Countries with confirmed cases",
             hover_data = ['Deaths'])
fig.show()

From the above we can see that top countries with most of the confirmed cases.

In [70]:
# new recovered count based on country
fig = px.bar(df[:20], x = 'Country/Region', y= 'New recovered',title="Top 10 Countries with newly recorved in a day",
             hover_data = ['Recovered'])
fig.show()

Top 20 countries with there recovery rate.

In [71]:
# Top countries death ration
fig = px.bar(df[:10], x = 'Country/Region', y= 'Deaths / 100 Cases',title="Top 10 Countries Deaths",
             hover_data = ['Deaths'])
fig.show()

From the above we can see that 10 countries with there death counts

In [72]:
# Finding the count of Deaths /100 Cases based on the country
fig = px.pie(df[:10], values='Deaths / 100 Cases', names='Country/Region', color='WHO Region',
             title='Death cases based on each country and region')
fig.show()

From the above we can see that count of death cases based on the country and region.

In [73]:
# Recorved cases based on Country/Region

fig = px.pie(df[:10], values='Recovered / 100 Cases', names='Country/Region', color='WHO Region',
             title='Recovered cases based on each country and region')
fig.show()

From the above we can see that the recovered cases based on each country and region.

In [74]:
# Recoved cases based on WHO region

fig = px.bar(df, y = "Recovered", x = "WHO Region", hover_data = ['Confirmed'], title='Recoverd cases per Region')
fig.show()

From the above we can see that the count of recovery for every WHO Region.

In [75]:
# Finding the active cases based on WHO Region
px.scatter(df[:20], y = "Active", x = "WHO Region", hover_data = ['Deaths'])

From the above we can see that active cases based on WHO Region

## Summary

From this we can get the following information:
1. confirmed cases based on the country and region.
2. Deaths based on the country and region.
3. Recovery based on the country and region.

# Thankyou